File: app/src/main/java/cat/xlagunas/andrtc/view/viewholder/FriendViewHolder.java
package cat.xlagunas.andrtc.view.viewholder;

import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import butterknife.BindView;
import butterknife.ButterKnife;
import cat.xlagunas.andrtc.R;
import cat.xlagunas.andrtc.view.adapter.FriendAdapter;
import xlagunas.cat.andrtc.domain.Friend;

/**
 * Created by xlagunas on 19/03/16.
 */
public abstract class FriendViewHolder extends RecyclerView.ViewHolder {
    @BindView(R.id.contact_username)
    TextView username;
    @BindView(R.id.contact_fullname)
    TextView fullName;
    @BindView(R.id.contact_thumbnail)
    ImageView thumbnail;

    public FriendViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    public void bind(final Friend friend) {
        this.username.setText(friend.getUsername());
        this.fullName.setText(this.itemView.getContext()
                .getString(R.string.friend_item_name, friend.getName(), friend.getSurname(), friend.getLastSurname()));

        Glide.with(this.itemView.getContext())
                .load(Uri.parse(friend.getThumbnail()))
                .placeholder(R.drawable.user_profile)
                .dontAnimate()
                .into(this.thumbnail);
    }

    protected FriendAdapter getFriendAdapter() {
        return (FriendAdapter) ((RecyclerView) itemView.getParent()).getAdapter();
    }
}

File: app/src/main/java/cat/xlagunas/viv/contact/viewholder/FriendViewHolder.kt
package cat.xlagunas.viv.contact.viewholder

import androidx.recyclerview.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import butterknife.BindView
import cat.xlagunas.domain.commons.Friend
import cat.xlagunas.viv.R
import cat.xlagunas.viv.contact.ContactListener
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions

abstract class FriendViewHolder(view: View, internal val contactListener: ContactListener) :
    androidx.recyclerview.widget.RecyclerView.ViewHolder(view) {

    @BindView(R.id.contact_thumbnail)
    lateinit var profileImage: ImageView

    @BindView(R.id.contact_fullname)
    lateinit var fullName: TextView

    @BindView(R.id.contact_username)
    lateinit var username: TextView

    internal lateinit var friend: Friend

    open fun bind(friend: Friend) {
        this.friend = friend
        fullName.text = friend.name
        username.text = friend.username

        Glide.with(itemView)
            .load(friend.image)
            .apply(RequestOptions.noAnimation())
            .apply(RequestOptions.placeholderOf(R.drawable.profile_placeholder))
            .apply(RequestOptions.circleCropTransform())
            .into(profileImage)
    }
}